vars/githubscm.groovy

```diff
@@ -93,11 +93,17 @@ def createBranch(String branchName) {
     println "[INFO] Created branch '${branchName}' on repo."
 }
 
-def commitChanges(String commitMessage, String filesToAdd = '--all') {
-    sh "git add ${filesToAdd}"
+def commitChanges(String commitMessage, Closure preCommit) {
```

Comment:
shouldn't be added a default `null` value for the preCommit argument like `Closure preCommit = null`

I'm guessing it because the `if(preCommit)` statement, somehow you are saying preCommit can be null, isn't it?

Reply:
good catch ...

Reply:
Removed the `if`, else it would conflict with the other `commitChanges` method and we don't want to break BC...